test_shaders.py

```diff
@@ -110,21 +110,36 @@ def print_msl_compiler_version():
         pass
 
 def msl_compiler_supports_version(version):
-    try:
-        subprocess.check_call(['xcrun', '--sdk', 'macosx', 'metal', '-x', 'metal', '-std=macos-metal' + version, '-'],
+    if float(version) >= 3.0:
```

Comment:
This is a bit too awkward. Duplicated code path is also not great. The function should just be:

```
msl_compiler_supports_version('-std=macos-metal2.0')
msl_compiler_supports_version('-std=macos-metal2.1')
msl_compiler_supports_version('-std=metal3.0')
```
That avoids the duplicated path and the bizarre float cast.

Reply:
good idea -- cleaner.